lib/ansible/modules/privileged_account_security/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: cyberark_authentication
+short_description: "Module for CyberArk Vault Authentication using Privileged Account Security Web Services SDK"
+author: "Edward Nunez (@enunez-cyberark)"
+version_added: "2.3"
+description:
+    - "Authenticates to CyberArk Vault using Privileged Account Security Web Services SDK and
+       creates a session fact that can be used by other modules. It returns an Ansible fact
+       called I(cyberarkSession). Every module can use this fact as C(cyberarkSession) parameter."
+
+
+options:
+    state:
+        default: present
+        choices: ['present', 'absent']
+        description:
+            - Specifies if an authentication logon/logoff and a cyberarkSession should be added/removed
+    userName:
+        required: False
+        description:
+            - The name of the user who will logon to the Vault
+    password:
+        required: False
+        description:
+            - The password of the user
+    newPassword:
```

Comment:
passing newPassword is a way to request a password change ... I added logic to everytime newPassword is sent to set changed to True... with newPassword it will report changes